binary/src/Cardano/Binary/FromCBOR.hs

```diff
@@ -476,10 +537,33 @@ decodeVector = decodeContainerSkelWithReplicate
   Vector.Generic.concat
 {-# INLINE decodeVector #-}
 
-instance (FromCBOR a) => FromCBOR (Vector.Vector a) where
+instance FromCBOR a => FromCBOR (Vector.Vector a) where
   fromCBOR = decodeVector
   {-# INLINE fromCBOR #-}
 
+instance FromCBOR a => FromCBOR (Seq.Seq a) where
+  fromCBOR = decodeSeq fromCBOR
+  {-# INLINE fromCBOR #-}
```

Comment:
Sometime I would love to learn from you about how you decide when to add the inline-ing.

Reply:
I add them, when I don't forget it. :smile: 

Small functions like that it is almost always safe to add an `INLINE`, larger functions however would be better with `INLINEABLE`. Occasionally it is better to use `NOINLINE` if you can see that inlining would result in duplicated work.

Most of the time it's a heuristic and therefore is hard to explain in concrete terms.